### PR TITLE
fix: show real open debts and gross coverage on Money Overview

### PR DESCRIPTION
**Pull request by christoph-teichmeister**
## Problem

The "Who needs to settle up" section on the Money Overview page was semantically wrong in three ways:

1. **Wrong data source**: It showed gross amounts paid by others for each person (raw `ChildTransaction` sums), not actual outstanding debts. This ignored the debt optimisation step entirely and showed numbers that did not match what people actually owe.
2. **Multi-currency bar scaling bug**: Both bar sections used `total_money_spent|first` to determine bar width, meaning all bars were scaled relative to the first currency only — wrong for rooms with mixed currencies.
3. **Settled debts not excluded**: The old query had no filter on `settled=False`, so paid-off debts still influenced the display.

### `money_spent_on_room_view.py`
- Renamed `money_owed_per_person_qs` → `money_covered_for_person_qs` (honest naming: gross consumption)
- Added `open_debts_per_person_qs`: queries the `Debt` model directly, filtered to `settled=False`, reflecting the actual optimised outstanding amounts
- Added `max_open_debt_per_currency`: dict used for correct per-currency bar scaling in the template

### `_money_spent_on_room.html`
- **"Who needs to settle up"** now shows two distinct sections:
  - **Open debt (after optimisation)** — green bars, scaled relative to max debt per currency
  - **Covered by others (gross)** — orange bars, scaled relative to total spent per currency
- **"Who covered the bills"**: bar scaling now matches per currency instead of using `|first`
- Chip in "Who covered the bills" now shows all currencies instead of just the first
- New CSS class `.graph-bar-tertiary` (orange gradient)
- Added `{% load core_tags %}`

### `core_tags.py`
- Added `get_item` template filter for dict key lookups in templates

### Tests
- Updated existing test to use renamed `money_covered_for_person_qs`
- Added `test_open_debts_per_person_qs_only_returns_unsettled`
- Added `test_max_open_debt_per_currency_reflects_max_unsettled`

## Business Impact

The Money Overview page now displays accurate financial information:
- **"Who needs to settle up" section** shows actual outstanding debts after debt optimization (rather than gross consumption amounts), giving users the true settlement requirements
- **Multi-currency support** fixed—bars and summaries now correctly scale per currency instead of only using the first currency, preventing misleading comparisons in diverse expense groups
- **Settled debts excluded** from open debt calculations, eliminating confusion about which debts actually require settlement

## Technical Changes

**View Layer (`money_spent_on_room_view.py`)**
- Renamed `money_owed_per_person_qs` → `money_covered_for_person_qs` to clarify that this represents gross consumption (not optimized debts)
- Added `open_debts_per_person_qs` context variable querying unsettled `Debt` objects, providing optimized outstanding amounts
- Added `max_open_debt_per_currency` dict to enable per-currency bar scaling

**Template Layer (`_money_spent_on_room.html`)**
- Restructured "Who needs to settle up" into two distinct sections:
  - "Open debt (after optimisation)" using green bars scaled per currency
  - "Covered by others (gross)" using orange bars for context
- Fixed "Who covered the bills" bar scaling to use per-currency maxima instead of single first-currency reference
- Updated all currency displays to show all applicable currencies instead of only the first

**Template Filter Support (`core_tags.py`)**
- Added `get_item` filter to enable template-level dictionary lookups, required for the per-currency scaling logic

## Testing

- Existing test updated to use the renamed `money_covered_for_person_qs`
- Added `test_open_debts_per_person_qs_only_returns_unsettled` to verify settled debts are properly filtered
- Added `test_max_open_debt_per_currency_reflects_max_unsettled` to validate per-currency scaling data

## Required Follow-up

- **Verification**: Test the Money Overview page in multi-currency scenarios to confirm bar scaling and chip displays are accurate
- **Edge cases**: Validate behavior when a user has no open debts (empty state should display cleanly)
- **Regression testing**: Confirm "Who covered the bills" section still functions correctly with the per-currency scaling changes

### CODE REVIEW

File: apps/debt/tests/test_views/test_money_spent_on_room_view.py [hunk 1]
```diff
@@ -64,10 +65,51 @@ def test_money_views_aggregate_totals_and_exclude_self_pay(self, room, guest_use
...
+        extra_user = UserFactory(name="Extra")
+        room.users.add(extra_user)
+
+        Debt.objects.create(debitor=guest_user, creditor=user, room=room, currency=currency, value=Decimal("30"), settled=False)
+        Debt.objects.create(debitor=extra_user, creditor=user, room=room, currency=currency, value=Decimal("10"), settled=False)
+
+        view = self._view_for_room(room)
+        max_map = view.max_open_debt_per_currency
+
+        assert max_map[currency.sign] == Decimal("30")
```
[the remark this answers is not in the document]
Comment (christoph-teichmeister) [reply apps/debt/tests/test_views/test_money_spent_on_room_view.py]: @coderabbitai done

File: apps/debt/tests/test_views/test_money_spent_on_room_view.py [hunk 2]
```diff
@@ -64,10 +65,51 @@ def test_money_views_aggregate_totals_and_exclude_self_pay(self, room, guest_use
...
+        assert open_debts[0]["debitor_name"] == guest_user.name
+        assert open_debts[0]["total_open_debt"] == open_debt.value
+
+    def test_max_open_debt_per_currency_reflects_max_unsettled(self, room, user, guest_user):
+        currency = room.preferred_currency
+        extra_user = UserFactory(name="Extra")
+        room.users.add(extra_user)
+
+        Debt.objects.create(debitor=guest_user, creditor=user, room=room, currency=currency, value=Decimal("30"), settled=False)
+        Debt.objects.create(debitor=extra_user, creditor=user, room=room, currency=currency, value=Decimal("10"), settled=False)
```
[the remark this answers is not in the document]
Comment (christoph-teichmeister) [reply apps/debt/tests/test_views/test_money_spent_on_room_view.py]: @coderabbitai done

File: apps/debt/views/money_spent_on_room_view/money_spent_on_room_view.py [hunk 3]
```diff
@@ -36,11 +37,41 @@ def total_money_spent(self) -> QuerySet[dict[str, object]]:
...
             self.get_base_queryset()
             .exclude(paid_for=F("parent_transaction__paid_by"))
             .values("paid_for__name", "parent_transaction__currency__sign")
-            .annotate(currency_sign=F("parent_transaction__currency__sign"), total_owed_per_person=Sum("value"))
+            .annotate(
+                currency_sign=F("parent_transaction__currency__sign"),
+                total_covered_for_person=Sum("value"),
+            )
             .order_by("paid_for__name")
         )
```
[the remark this answers is not in the document]
Comment (christoph-teichmeister) [reply apps/debt/views/money_spent_on_room_view/money_spent_on_room_view.py:52]: @coderabbitai done

File: apps/debt/views/money_spent_on_room_view/money_spent_on_room_view.py [hunk 4]
```diff
@@ -36,11 +37,41 @@ def total_money_spent(self) -> QuerySet[dict[str, object]]:
...
+                currency_sign=F("currency__sign"),
+                total_open_debt=Sum("value"),
+            )
+            .order_by("debitor__name")
+        )
+
+    @context
+    @property
+    def max_open_debt_per_currency(self) -> dict[str, object]:
+        """Max open debt value per currency – used for bar scaling in the template."""
```
[the remark this answers is not in the document]
Comment (christoph-teichmeister) [reply apps/debt/views/money_spent_on_room_view/money_spent_on_room_view.py]: @coderabbitai done

File: apps/transaction/templates/transaction/partials/_money_spent_on_room.html [hunk 5]
```diff
@@ -165,34 +165,63 @@ <h5 class="mb-0">{% trans "Who needs to settle up" %}</h5>
...
                                     </div>
                                 {% endif %}
-                            {% empty %}
-                                <div class="empty-state mt-3">
-                                    <p class="fw-semibold mb-1">{% trans "Nothing owed right now" %}</p>
-                                    <p class="text-muted small mb-0">{% trans "All participants are even for the tracked expenses." %}</p>
-                                </div>
                             {% endfor %}
-                        {% endwith %}
+                        {% endif %}
```
[the remark this answers is not in the document]
Comment (christoph-teichmeister) [reply apps/transaction/templates/transaction/partials/_money_spent_on_room.html:223]: @coderabbitai done

File: apps/debt/tests/test_views/test_money_spent_on_room_view.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (christoph-teichmeister) [reply apps/debt/tests/test_views/test_money_spent_on_room_view.py]: @coderabbitai done

File: apps/debt/tests/test_views/test_money_spent_on_room_view.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (christoph-teichmeister) [reply apps/debt/tests/test_views/test_money_spent_on_room_view.py]: @coderabbitai done

File: apps/debt/views/money_spent_on_room_view/money_spent_on_room_view.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (christoph-teichmeister) [reply apps/debt/views/money_spent_on_room_view/money_spent_on_room_view.py:52]: @coderabbitai done

File: apps/debt/views/money_spent_on_room_view/money_spent_on_room_view.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (christoph-teichmeister) [reply apps/debt/views/money_spent_on_room_view/money_spent_on_room_view.py]: @coderabbitai done

File: apps/transaction/templates/transaction/partials/_money_spent_on_room.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (christoph-teichmeister) [reply apps/transaction/templates/transaction/partials/_money_spent_on_room.html:223]: @coderabbitai done
